classify.py

```diff
@@ -313,58 +335,121 @@ def evaluate_samesource(desc, dataset, n_frequent_words, tokens_per_sample, max_
     LOG.info(f'{desc}: number of speakers: {np.unique(y).size}')
     LOG.info(f'{desc}: number of instances: {y.size}')
 
-    X = preprocessor.fit_transform(X)
+    X = preprocessor.fit_transform(X) # shouldn't this take place on the x_train and then on x_test?
```

Comment:
Is it possible to leak information from test data to training this way?

Reply:
done